File: Overlays/ingame/src/data/stateData.ts
import FrontEndObjective from "./frontEndObjective";
import GoldEntry from "./goldEntry";
import InfoSidePage from "./infoSidePage";
import Inhibitor, { InhibitorInfo } from "./inhibitor";
import Objective from "./objective";
import ScoreboardConfig from "./scoreboardConfig";

export default class StateData {
    dragon: FrontEndObjective;
    baron: FrontEndObjective;

    gameTime: number;
    gamePaused: boolean;

    blueGold: number;
    redGold: number;

    goldGraph: GoldEntry[];
    inhibitors: InhibitorInfo;

    scoreboard: ScoreboardConfig;
    infoPage: InfoSidePage;

    blueColor: string;
    redColor: string;

    constructor(message: any){
        this.dragon = message.dragon;
        this.baron = message.baron;
        this.gameTime = message.gameTime;
        this.gamePaused = message.gamePaused;
        this.blueGold = message.blueGold;
        this.redGold = message.redGold;
        this.goldGraph = [];
        if(message.goldGraph != undefined && message.goldGraph != null) {
            for(const [key,value] of Object.entries(message.goldGraph)) {
                this.goldGraph.push(new GoldEntry(+key, value as number ))
            }
        }
        this.inhibitors = message.inhibitors;
        this.scoreboard = new ScoreboardConfig(message.scoreboard);
        this.infoPage = new InfoSidePage(message.infoPage);

        this.blueColor = message.blueColor;
        this.redColor = message.redColor;
    }
}

File: LeagueBroadcast/Ingame/State/StateData.cs
using LeagueBroadcast.Common;
using LeagueBroadcast.Common.Controllers;
using LeagueBroadcast.Ingame.Data.Frontend;
using LeagueBroadcast.Ingame.Data.LBH;
using LeagueBroadcast.Ingame.Data.LBH.Objectives;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace LeagueBroadcast.Ingame.State
{
    public class StateData
    {

        #region NotSerialized
        [JsonIgnore]
        public BackEndObjective backBaron;

        [JsonIgnore]
        public BackEndObjective backDragon;

        private State gameState;
        #endregion 

        public FrontEndObjective dragon;

        public FrontEndObjective baron;

        public double gameTime;
        public bool gamePaused;

        public float blueGold => gameState.blueTeam.GetGold(gameTime);
        public float redGold => gameState.redTeam.GetGold(gameTime);

        public Dictionary<double, float> goldGraph => gameState.GetGoldGraph();
        public InhibitorInfo inhibitors;

        public ScoreboardConfig scoreboard;

        public InfoSidePage infoPage;
        public string blueColor => gameState.blueTeam.color;
        public string redColor => gameState.redTeam.color;

        public StateData()
        {
            this.gameState = BroadcastController.Instance.IGController.gameState;
            this.dragon = new (Objective.ObjectiveType.Dragon, 300);
            this.baron = new (Objective.ObjectiveType.Baron, 1200);
            this.backBaron = new(0);
            this.backDragon = new(0);
            this.scoreboard = new ScoreboardConfig();
            this.inhibitors = new InhibitorInfo();
        }

        public bool ShouldSerializegoldGraph()
        {
            return IngameController.CurrentSettings.GoldGraph;
        }

        public bool ShouldSerializeinhibitors()
        {
            return IngameController.CurrentSettings.Inhibs;
        }

        public bool ShouldSerializebaron()
        {
            return IngameController.CurrentSettings.Baron && backBaron.DurationRemaining > 0;
        }

        public bool ShouldSerializedragon()
        {
            return IngameController.CurrentSettings.Elder && backDragon.DurationRemaining > 0;
        }

        public bool ShouldSerializeinfoPage()
        {
            return IngameController.CurrentSettings.SideGraph;
        }
    }
}
